sw/device/lib/testing/usb_logging.c

```diff
@@ -0,0 +1,549 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/usb_logging.h"
+
+#include <ctype.h>
+
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/pinmux_testutils.h"
+#include "sw/device/lib/testing/usb_testutils.h"
+#include "sw/device/lib/testing/usb_testutils_controlep.h"
+
+#define MODULE_ID MAKE_MODULE_ID('u', 's', 'l')
+
+// Size of each logging buffer; data is collected in a buffer until the buffer
+// is full (or a timeout occurs) and is then transmitted as a single transfer
+// at higher throughput.
+enum {
+  kUSBLogBufferSize = 0x800u,
```

Comment:
Nit. We normally put all the constants inside the same enum block, with comments.